src/import/import.rs

```diff
@@ -56,21 +58,38 @@ impl<Client: ImportClient> ImportJob<Client> {
         let start = Instant::now();
         info!("start"; "tag" => %self.tag);
 
+        // Join and check results.
+        let mut res = Ok(());
+        let mut retry_ranges = Arc::new(Mutex::new(Vec::new()));
         // Before importing data, we need to help to balance data in the cluster.
         let job = PrepareJob::new(
             self.cfg.clone(),
             self.client.clone(),
             Arc::clone(&self.engine),
         );
-        let ranges = job.run()?;
-        let handles = self.run_import_threads(ranges);
-
-        // Join and check results.
-        let mut res = Ok(());
-        for h in handles {
-            if let Err(e) = h.join().unwrap() {
-                res = Err(e)
+        for i in 0..MAX_RETRY_TIMES {
+            let ranges = if i == 0 {
+                job.run()?.drain(0..).map(|range| range.range).collect()
```

Comment:
It's not what I mean. What I mean is 
```rust
let mut ranges = PrepareJob::new(...).run()?;
for i in 0..MAX_RETRY_TIMES {
    let mut retry_ranges = Arc::new(Mutex::new(Vec::new()));
    let handles = self.run_import_threads(ranges, Arc::clone(&retry_ranges));
    for h in handles {
        if let Err(e) = h.join().unwrap() {
            res = Err(e)
        }
    }
    let retry_count = retry_ranges.lock().unwrap().len();
    if retry_count < 1 {
        break;
    }
    ranges = retry_ranges.drain(..).map(|range| range.range).collect();
}
```

Reply:
`PrepareJob.run` returns `RangeInfo` but `retry_ranges` is a vector of ranges.

Reply:
seems just need to some make some tiny changes to fix that, like this?
```rust
let mut ranges = PrepareJob::new(...).run()?.into_iter().map(|range| range.range).collect();
for i in 0..MAX_RETRY_TIMES {
    let mut retry_ranges = Arc::new(Mutex::new(Vec::new()));
    let handles = self.run_import_threads(ranges, Arc::clone(&retry_ranges));
    for h in handles {
        if let Err(e) = h.join().unwrap() {
            res = Err(e)
        }
    }
    let retry_count = retry_ranges.lock().unwrap().len();
    if retry_count < 1 {
        break;
    }
    ranges = retry_ranges.drain(..).collect();
}
```
Furthermore, seems that we do not need `RangeInfo` from `PrepareJob.run`, we can just return `Range` instead of `RangeInfo`

Reply:
addressed